EIPS/eip-4519.md

```diff
@@ -0,0 +1,239 @@
+The user sends PK<sub>UD</sub> signed to the asset. The latter checks the signature and optionally checks PK<sub>UD</sub> on the attribute dataEngagement of its token. If the user is verified, the asset calculates:
+
+K<sub>UD</sub> = SK<sub>DEV</sub>*PK<sub>UD</sub>:
+
+If everything is correctly done, K<sub>U</sub> and K<sub>UD</sub> are the same since:
+ 
+K<sub>U</sub>=PK<sub>DEV</sub>*SK<sub>UD</sub>=(SK<sub>DEV</sub>*P)*SK<sub>UD</sub>= SK<sub>DEV</sub>*(SK<sub>UD</sub>*P)=SK<sub>DEV</sub>*PK<sub>UD</sub>
+
+Using the function userEngagement, the asset sends the K<sub>UD</sub> obtained and if it is the same as K<sub>U</sub>, then the state of the token changes to "Engaged with user" and the event UserEngaged is sent. Once the asset receives the event, it changes its operation mode to "Engaged with user". This process is shown in Figure 5. From this moment, the asset can be managed by the user. 
+ 
+ ![Figure 5: Steps in a successful user and asset mutual authentication process](../assets/eip-4519/images/Fig5_rev.png)
+
+Since the establishment of a shared secret is very important for a secure communication, we propose the inclusion of the attributes hashK_OD, hashK_UD, and dataEngage in the SmartNFT. The first two attributes define, respectively, the hash of the secret shared between the asset and its owner and between the asset and its user. Assets, owners, and users should check they are using the correct shared secrets. The attribute dataEngagement defines the public data needed for the agreement. If the mutual authentication fails, dataEngagement allows detecting which parts failed.
+  
+## Backwards Compatibility
+This implementation is an extension of the ERC-721 standard, then it is not only compatible with the standard, but also is a way to improve some of the current NFT tokens based on the ERC-721 standard.
+
+## Test Cases
+The test case presented in the paper shown below is addressed in **0x7eB5A03E7ED70ABf70fee48965D0411d37F335aC** and the code is available [here](../assets/eip-4519/PoC_SmartNFT/README.md)
+
+## Reference Implementation
+A first version was presented in a paper of the Special Issue **Security, Trust and Privacy in New Computing Environments** of **Sensors** journal of **mdpi** editorial. The paper, entitled [Secure Combination of IoT and Blockchain by Physically Binding IoT Devices to Smart Non-Fungible Tokens Using PUFs](https://doi.org/10.3390/s21093119), was written by the same authors of this draft.
```

Comment:
It looks like there's an implementation in the `assets` directory. Might be worth linking to that.

We also no longer allow external links, so if the license allows it, it might be worth including the paper in the assets directory.